### DISCUSSION TITLE
Can we expect M1 support any time?

### DISCUSSION THREAD

**Discussion by joeyqleq** [chris1111/Wireless-USB-Big-Sur-Adapter#128] [General]
Ive spent 4 hours trying several workarounds but these drivers and the other D-link driver simply wont work on Big sur running on M1 - super frustrating!

Anyone know something i dont know?

**Comment by chris1111** [comment]
I dont have a M1 Mac to test it so thats the problem
But what is the reason for wanting a USB adapter when the integrated WIFI of M1 mac chip is ultra powerful?

**Reply by joeyqleq** [reply]
is there i can help out with testing dev on my m1?

**Reply by wangmauler** [reply]
> I dont have a M1 Mac to test it so thats the problem But what is the reason for wanting a USB adapter when the integrated WIFI of M1 mac chip is ultra powerful?

My mother has an M1 that is used in a vertical dock so she can use an external monitor/keyboard for her poor eyesight. While in the dock, the laptop is closed and the wifi signal goes from ~75mbps down to about 1mbps. Taking it out of the dock brings it up to about 20mbps, and opening it (still in exactly the same spot) brings it back up to about 75mbps. 

I need to use an external WiFi so she can watch her Sewing Tutorials on youtube.

I'm willing to help with testing, as I have many years of experience in IT in windows, a few years in linux and on MAC.

-I've disabled SIP and disabled gatekeeper by booting into recovery and using terminal.
-I can install this program, but it does not seem to allow installation of the realtek driver. The first time I installed, the computer said it could not install the realtek package as it was not an authorized app, but in the background your program continued to install.
-Your program installed, but no adapters are found.

I have a TP-WN722Nv2

**Reply by sebastianpulido** [reply]
The reason is to share you wifi connection to e.g. an Xbox or any other device from e.g. a hotel which won't let you connect such devices.
Macs do not let you share your Wifi connection if you are at the same time connected to a Wifi network, therefore the need for an usb wifi adapter

**Comment by chris1111** [comment]
Do you have install [Roseta 2](https://www.computerworld.com/article/3597949/everything-you-need-to-know-about-rosetta-2-on-apple-silicon-macs.html)

**Reply by joeyqleq** [reply]
No if never occurred to me to get it because most of not all developers have adapted their software to arm. Should I install if? U think ur tool would work this way? I needed support for a Tenda W311U+ (RT2870) usb dongle because I installed kali Linux as a virtual machine and needed a usb single for sniffing WiFi packets and reconnect with “what’s new” with the computer security world because I the USB adapter lying around somewhere so I thought to myself ... why not?! But im not going to go as far as to purchase once that the M1 supports just go feed my 1-2 day hobby. So, u think I should install Rosetta 2?

**Comment by chris1111** [comment]
Of course without Roseta I will never works

**Comment by joeyqleq** [comment]
ok as it turns out it was already installed

**Comment by blasphemousjohn** [comment]
Hi @chris1111 & @joeyqleq, I am not fully sure that #48 confirms that the M1 macs cannot run the software.  I am going to attempt to run it after disabling SIP.  The problem that I was having with the _stock_ Netgear A7000 drivers was that the System Pref's 'Security' would not prompt me to 'Allow' the drivers (extensions) to work, after flashing the 'System Extension Blocked' message during install.  My gut instinct is that this was caused by not having SIP disabled as Big Sur deprecated these old 'kext' extensions.  

I will try again and update this post.  @chris1111 let me know if there is any other testing I can do for you, as I have available an M1 mac.

**Reply by joeyqleq** [reply]
I’m having trouble disabling Big Sur’s security. I was trying to install the kext file from the package manually issuing a kext utility but whenever I disable FileVault encryption then boot into recovery to disable SIP and authenticated-root I get stuck in a reboot loop every time I log into Big sur so me as well I’m not able to manually install the kexts. I’m only able to login if I run the security utility in recovery to re-enable SIP... I don’t know y big sur on M1 MacBook Air crashes when these settings are disable so I’m not able to fully gauge wether a manual install of the kexts will work . Anyone have any idea why disabling SIP and Authenticated Root on big sur gets me stuck in a boot loop?

**Reply by blasphemousjohn** [reply]
@joeyqleq I too have an M1 Macbook Air, however I did not have any problems disabling SIP?  Can you confirm the steps that you are doing to disable SIP?  

As a side note, after disabling SIP and attempting to install the drivers I am greeted by the 'System Extension Blocked' message, however there is still no prompt in 'System Preferences' -> 'Security' to 'allow' the install to proceed, and thus I am unable to get the drivers to work.  I have confirmed SIP disabled status.  As of now I have not found any solution to get this to work on M1 Macs.

**Reply by dariushde** [reply]
> @joeyqleq I too have an M1 Macbook Air, however I did not have any problems disabling SIP? Can you confirm the steps that you are doing to disable SIP?
> 
> As a side note, after disabling SIP and attempting to install the drivers I am greeted by the 'System Extension Blocked' message, however there is still no prompt in 'System Preferences' -> 'Security' to 'allow' the install to proceed, and thus I am unable to get the drivers to work. I have confirmed SIP disabled status. As of now I have not found any solution to get this to work on M1 Macs.

Any updates? I still haven't found a solution to make this work on M1 macbooks.

**Reply by blasphemousjohn** [reply]
Sorry for the late response.  Legacy kext extensions need to be re-written for M1 Macs (they will not work even through Rosetta 2), this means it is up to the big companies to update their drivers to support M1.  I suggest contacting adapter makers like Netgear etc and pushing them to update drivers for M1 macs.  Netgear has stated that they are working on an update for their A7000 series wireless adapters, with an original anticipated release of Q3 2021.  This was pushed about a week or two ago to a now unannounced (TBD) date.  Seems they (Netgear and other hardware manufacturers) may be waiting on Realtek (correct me if I am wrong?) for the actual driver.

**Comment by chris1111** [comment]
@blasphemousjohn You Probably right

**Comment by sebastianpulido** [comment]
In the meantime,
you can use VMWare for M1 Mac and run the driver below in Ubuntu ARM:
https://github.com/sebastianpulido/rtl8821CU

It is working for me flawlessly.
Note: Parallels 17.0 also works, but it does not work with Parallels 17.1

**Comment by Surj76** [comment]
Hi Chris,

I have a alfa awus036ach and am using a MacBook Pro M1.
I would love to have drivers working for this too - as I would like to use this wifi adapter with WiFi Explorer Pro 3.

Its a useful tool to help me troubleshoot wireless networks, and the M1 pro is my main machine of use.

I have followed the steps outlined in the fork - disabled SIP and the install has worked - however there are some errors:

Mar  6 00:58:18 Installer[3237]: Could not load resource license: (null)
Mar  6 00:58:27  installd[1199]: PackageKit: ----- Begin install -----
Mar  6 00:58:29  installd[1199]: PackageKit: PKInformSystemPolicyInstallOperation failed with error:An error occurred while registering installation with Gatekeeper.
Mar  6 00:58:41  installd[1199]: Could not open /Private/tmp/RtWlanHelper for AOT translation
Mar  6 00:58:41  installd[1199]: AOT translation failed for /Library/Extensions/RtWlanU1827.kext/Contents/MacOS/RtWlanU1827
Mar  6 00:58:41  installd[1199]: AOT translation failed for /Library/Extensions/RtWlanU.kext/Contents/MacOS/RtWlanU
Mar  6 00:58:41  installd[1199]: PackageKit: ----- End install -----

When trying to load the kext manually I do get some errors - and these errors make complete sense as the kext files provided are designed for Intel:

sudo kextutil -t -v /Library/Extensions/RtWlanU1827.kext
Password:
Defaulting to kernel file '/System/Library/Kernels/kernel'
Executing: /usr/bin/kmutil load --bundle-path /Library/Extensions/RtWlanU1827.kext
Error Domain=KMErrorDomain Code=71 "Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU1827 v1827.4.b36 in executable kext bundle com.realtek.driver.RtWlanU1827 at /Library/Extensions/RtWlanU1827.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU1827 v1827.4.b36 in executable kext bundle com.realtek.driver.RtWlanU1827 at /Library/Extensions/RtWlanU1827.kext" UserInfo={NSLocalizedDescription=Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU1827 v1827.4.b36 in executable kext bundle com.realtek.driver.RtWlanU1827 at /Library/Extensions/RtWlanU1827.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU1827 v1827.4.b36 in executable kext bundle com.realtek.driver.RtWlanU1827 at /Library/Extensions/RtWlanU1827.kext}

Please please please - if you can, please provide an installer which is compatible with M1 Pro as there are limited options for external wireless adapters.
This particular adapter I am using works with most applications I already have.

**Reply by hlainez** [reply]
Any update on this?

**Comment by EnumSet** [comment]
It will not work on M1 ((

**Comment by chris1111** [comment]
Why using a cheapest wifi dongle on most powerfull wifi in the laptop world 

![2670155727](https://user-images.githubusercontent.com/6248794/169667835-64635225-c513-4d57-ac1e-452fa8dd3a43.gif)![1649127375](https://user-images.githubusercontent.com/6248794/169667841-8f44b8bf-e67a-417b-9608-6eb72a65e326.gif)

**Reply by wangmauler** [reply]
In my case, my M1 MacBook is docked in a vertical mount, next to the monitor. When the MacBook is physically closed as in my case, the wifi signal drops more than 80% and simple web pages take a long time or time out. This is why I require a simple wifi dongle. If Apple had engineered their laptops better this would be a non-issue.

**Reply by TG-Fluke** [reply]
There is also the use-case where you need to use Internet-sharing over WiFi -- which requires a second WiFi port.  I use this when operating from a marina (with my Intel-based Mac).  The Alfa USB WiFi adapter picks up the distant WiFi signal much better than my Intel Mac, then the WiFi in my Mac serves as the local WiFi for other devices near me to share the distant connection.

Being able to address this scenario is one of a couple of issues I need to solve before moving to an M1-based mac.

**Reply by rfors** [reply]
> Why using a cheapest wifi dongle on most powerfull wifi in the laptop world

First, thanks for your continued efforts on this! To answer your question, in my use case it's because I need to connect to multiple WiFi-enabled cameras at the same time and control / stream from them, and the SDK is extremely limited on non-Mac. Decent but cheap dongles get it done.

**Reply by jamall007** [reply]
Because the booster and antennae I have on my roof allow me to "access" more than fifty wifi networks within a 3km radius, only one or two of which my M1 Mac ever gets an occasional fleeting glimpse of. Just like Netscape Navigator, Hotline and Carracho, paying for internet bandwidth is but a quaint memory from the distant past with zero probability of returning.

**Reply by barrettlyon** [reply]
I use it to repeat wifi on airlines so I can share it to my wife and kids. Super helpful.

**Reply by ballo** [reply]
I need an external wifi adapter because I get zero signal where I use my MacBook.

**Reply by Overdown26** [reply]
@chris1111 In my case, my mac mini w/m1 wifi module is defective, if I turn it on bluetooh and wifi goes on and off and is impossible to use, so I have on lan all the time.

**Comment by chris1111** [comment]
**Use all steps here for M1**

Yes its possible 😀  Download ➤ [Notarized Wireless USB Big Sur Adapter.zip](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/files/7713890/Notarized.Wireless.USB.Big.Sur.Adapter.zip)

Followed the guide here

Make your Monterey or Big Sur is SIP Enable
Double clic to **Consent.command**
Result ⇩
![Screen Shot 2021-12-15 at 7 51 40 AM](https://user-images.githubusercontent.com/6248794/146191467-de1b0805-1af7-4237-ad5f-d075e1c57e9d.png)


#### Boot in Recovery open Utility / Terminal

#### Type: the command bellow followed by Enter
```bash
csrutil disable
```
#### Type: the command bellow followed by Enter
```bash
/usr/sbin/spctl kext-consent add ZYM2ETK3E7
```
Closed Terminal then reboot
- Followed the guide ➤ [Wireless USB Big Sur Adapter](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter)
Install - Wireless USB Big Sur Adapter.app
Boot in Recovery open Utility / Terminal Type: the command bellow followed by Enter
`csrutil disable` then 
`csrutil enable --without kext`

Closed Terminal then reboot
Enjoy!

**Reply by pokhachevskiy** [reply]
anyone has worked? 
Macbook Air M1 11.6.2 - not helped, neither SIP disabled or enable --without kext

**Reply by sebastianpulido** [reply]
In the meantime,
you can use VMWare/Parallels for M1 Mac and run this driver in Ubuntu ARM:
https://github.com/sebastianpulido/rtl8821CU

**Reply by mccruz** [reply]
I was able to get the software installed. The icon at the top nav bar was showing but it couldn't find the usb adaptor that I had plugged in. During setup, confirmed that SIP was disabled. Tried using this with TP-LINK Archer T3U.

**Reply by bprodoehl** [reply]
These steps are also not working for me on an M1 MacBook running Monterey (12.4).  All the csrutil and spctl stuff went fine, Consent.command finishes with the success messages as posted above, but the kext is an x86_64 bundle, so it refuses to load. From my understanding, an x86_64 kext bundle will not work on an arm64e kernel. Is there an arm64e kext bundle available?

Here's what I get running `file` against the kexts in the notarized driver zip linked to above:

```
$ file drivers/RtWlanU.kext/Contents/MacOS/RtWlanU 
drivers/RtWlanU.kext/Contents/MacOS/RtWlanU: Mach-O 64-bit kext bundle x86_64
```

And here's what I get when I try to load the driver manually, after following all the steps to install it and the application:

```
$ sudo kextload /Library/Extensions/RtWlanU.kext 
Executing: /usr/bin/kmutil load -p /Library/Extensions/RtWlanU.kext
Error Domain=KMErrorDomain Code=71 "Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU v1830.32.b27 in executable kext bundle com.realtek.driver.RtWlanU at /Library/Extensions/RtWlanU.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU v1830.32.b27 in executable kext bundle com.realtek.driver.RtWlanU at /Library/Extensions/RtWlanU.kext" UserInfo={NSLocalizedDescription=Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU v1830.32.b27 in executable kext bundle com.realtek.driver.RtWlanU at /Library/Extensions/RtWlanU.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU v1830.32.b27 in executable kext bundle com.realtek.driver.RtWlanU at /Library/Extensions/RtWlanU.kext}
```

**Comment by boomlex** [comment]
Same issue over here.
Not working on **Macbook Pro M1**.

I have some issues with the stock Wifi card which Apples refused to replace because they don't find to have any defects.
I only can connect to some wifi networks. I already send them the logs from the connections but still refuse to believe me no matter the warranty. Other Macbook Pro M1 doesn't have this issue like mine, we already tested out. 

The only workaround which I was hoping to work is with external wifi adapter and this is why I buy the TP-Link Archer T2U with usb-c to usb cable, until I found that it will not detect. 

Tried everything in this thread but still nothing and if I run it manually like @bprodoehl I am receiving the exact same result from the terminal.

I hope there will be a solution for future for M1 chips.

Keep up the good work @chris1111 we really appreciate your work and all other contributions to the project.

**Reply by IzsakiRobi** [reply]
Hi!

First: Big and Good work, helped some people in the world 😊

But please recompile the kexts with 64bit Intel and ARM in xcode.
Me and Some people really want this... 😊 

Thank You!

**Reply by chris1111** [reply]
> Hi!
> 
> First: Big and Good work, helped some people in the world 😊
> 
> But please recompile the kexts with 64bit Intel and ARM in xcode. Me and Some people really want this... 😊
> 
> Thank You!


Gives me the source code I will recompile them

**Reply by bprodoehl** [reply]
I understand that you didn't compile the kexts, and thanks so much for this project, the notarized drivers make things so much easier on Intel-based Macs. I wasn't sure if the comments about it working on Apple Silicon were just not right, or if I had the wrong download link for the kext.

**Reply by IzsakiRobi** [reply]
The notarized kexts also not work in apple silcon 😞
Manual load i get this:

_Error Domain=KMErrorDomain Code=71 "Incompatible architecture: Binary is for x86_64, but needed arch arm64e
Unsupported Error: one or more extensions are unsupported to load 	Kext com.realtek.driver.RtWlanU v1830.32.b27 in executable kext bundle com.realtek.driver.RtWlanU at /Users/izsakirobi/Downloads/Notarized Wireless USB Big Sur Adapter/drivers/RtWlanU.kext_

**Reply by chris1111** [reply]
> I understand that you didn't compile the kexts, and thanks so much for this project, the notarized drivers make things so much easier on Intel-based Macs. I wasn't sure if the comments about it working on Apple Silicon were just not right, or if I had the wrong download link for the kext.


It work on Intel Mac , Already testing by me and manny other
[The Notorized package](https://github.com/chris1111/Wireless-USB-OC-Big-Sur-Adapter/discussions/167)

**Comment by boomlex** [comment]
Do we have some update for M1 chips? Is there a chance that they will be supported soon?

**Comment by sodansoap** [comment]
I’ve just ditched a work Lenovo and am now using a new 16” M1 MacBook Pro running 12.something Monterey and ideally need the primary Apple WiFi NIC to connect to my FlightScope golf launch monitor as it irritatingly has a WiFi only connection. Additionally, various Golf Sim applications require internet resources to operate that I need my TPLink Archer T2U nano USB Adaptor for (or the other way around, whatever)

I did try a few things, disabling SIP and using that custom Big Sur installer floating around - mainly following this Brazilian YouTube guy (muted as for me the audio is totally pointless) https://youtu.be/IRid_anETtI but I can’t seem to get the ESP Mount application to mount a flash drive properly - he’s got a bunch of files but I’m not really understanding where they come from - I assume he does explain but I don’t understand Portuguese. When he opens his ESP Mount he’s got a few devices showing up I don’t have (one is his flash drive) - unsure about the 2nd one and he appears to copy files from one of them to his flash mount and then reboots and boots into his flash that then does something and forwards to booting back into MacOS. It *appears* to be the most promising workaround online from my searches and frustrating I can’t work it out.

I’ve given up pissing around for now - thinking an Apple Thunderbolt to RJ45 Ethernet adapter plugged into a Wi-Fi extender configured to connect to my router. Just annoying as it means I can’t roam around using batteries as the WiFi extender obviously requires mains power which for me means messing around setting up extension reels run from a shed

I also tried to use internet resources via my iPhone hotspot but can’t for the life of me get my iPhone paired via Bluetooth - it connects briefly and a second later shows “not connected” - another unexplained struggle, ridiculous as Apple products should be totally seamless and partly the reason I moved to Mac anyway!

Let me know if any Portuguese speakers follow that video properly and has any luck - he does seem to get it working! 😆

**Reply by friscoMad** [reply]
That video instructions are for installing into a hackingtosh with OpenCore not a real M1.
As stated before we need a kernel extension compiled with arm64 support but that will not happen unless someone reverse-engineers the current kernel extensions and transpile them to arm84 or new drivers are provided by the manufacturers.

**Comment by danielktdoranie** [comment]
Just posting this here to let other people know this **DOES NOT** work for me. I am on:

MacBook Air (M1, 2020)
macOS "Ventura" Version 13.1 (22C65)
Apple's official USB-C to USB Adapter
Alfa AWUS1900 (Realtek RTL8814U Chipset)

Read through this discussion about 4 times, read the readme, watched the video (did all the steps in the video), even tried the updated instructions regarding "Notarized Wireless USB Big Sur Adapter.zip" from @chris1111 

**Sadly it  doesn't work.** 

I think my solution will be to just use this adapter on my Thinkpad tuning Kali, where I know it works.

**Reply by richardokonicha** [reply]
doesn't work on silicon

**Comment by AlexeyInwerp** [comment]
after i fixed heavily waterdamaged 2020 M1 pro,  unfortunately i was not able to restore WiFi. Unlike prior models, this chip is married to the CPU, so replacement would not work. 
WIFI chip deaths is really common issue on 13" pro (due to component placement it gets electrocuted by 1 liquid drop sucked from display edge). I figured how to resolve partial short on some devices, but unfortunately some parts of the IC remain damaged and it can't connect to any network. This is really frustrating that you cant untie CPU from WIFI chip and it is not possible to use USB wireless adapter of any sort (i tried plenty of them including ones which worked fine with intel macs)

Update:
For now, the cheapest and easiest option is to use mini Wi-Fi repeater like TL-WR902AC to connect it with USBC-HUB-Ethernet. Second option is USB Tethering on iphone/android. It seems we wont see any solution for Wifi module replacement and still no sign of driver support for ARM macs.

**Comment by chris1111** [comment]
Did you see and try [THIS](https://www.sweetwater.com/sweetcare/articles/kernel-extensions-on-mac-with-apple-silicon/)

**Reply by bombruch** [reply]
I couldn't open your link so I tried to google "kernel-extensions-on-mac-with-apple-silicon". Used this link: https://macsupport.tuxera.com/hc/en-gb/articles/4409208805522-How-do-I-enable-third-party-kernel-extensions-on-Apple-silicon-based-Macs


Low level was selected when got there. So I just tapped on  "Allow user management of kernel extensions from identified developers". Screen in Russian but it gives overall picture:
![IMG_5596](https://github.com/user-attachments/assets/0aff5b14-5b25-4593-8532-8b7ce84ff945)


Gate keeper profile status:
<img width="715" alt="Screenshot 2025-02-21 at 16 07 09" src="https://github.com/user-attachments/assets/b89f361a-30d3-41c3-9386-03c48d941ed9" />
<img width="495" alt="Screenshot 2025-02-21 at 16 07 19" src="https://github.com/user-attachments/assets/bd02afd7-8426-438a-ba1a-efbb015ecea1" />


Sip status:
<img width="498" alt="Screenshot 2025-02-21 at 16 07 33" src="https://github.com/user-attachments/assets/4b9b4551-96eb-44a7-9afa-1597b5bbf164" />


After another installation nothing appears in Privacy & Security:
<img width="705" alt="Screenshot 2025-02-21 at 16 16 26" src="https://github.com/user-attachments/assets/de589906-f64f-4b7d-9297-d0ca57ca9298" />


Setup looks like:
MAC BOOK:
<img width="283" alt="Screenshot 2025-02-21 at 16 42 24" src="https://github.com/user-attachments/assets/588f6061-08e0-4988-9444-09f65df195de" />


Adapter from usb to usb + Adapter USB WiFi:
![IMG_5597](https://github.com/user-attachments/assets/18a2cfd3-27c2-4319-8991-ce016120dffa)
Chip 8812BU
![IMG_5604](https://github.com/user-attachments/assets/f2c88ddb-f893-4b55-b16d-e85fd87af42b)


Just in case: I have old Mac book from 2014 and followed by steps in your guide and everything works with this USB WIFI adapter:
![IMG_5600](https://github.com/user-attachments/assets/73babd2c-158a-4f6c-a98b-2612e5d4cc39)


Any ideas? Any paths to check? Any commands to run?

**Reply by bombruch** [reply]
Read carefully thread - seems like need to recompile kexts... 
<img width="1725" alt="Screenshot 2025-02-21 at 17 55 03" src="https://github.com/user-attachments/assets/b4d8aa42-0f04-4b1e-b3ad-150e493fa3f9" />

**Comment by ballo** [comment]
Is there anything in this driver, firmware or otherwise, which in theory would make this x86 only? I ask before I even attempt a aarch64 compile.

**Reply by chris1111** [reply]
> Is there anything in this driver, firmware or otherwise, which in theory would make this x86 only? I ask before I even attempt a aarch64 compile.

If you have the source of the kext to recompile post here the source

**Comment by AlexeyInwerp** [comment]
wifi chip replacement for those who have failed wifi will be available soon for macbook repairs. most likely in comming months, it would require some BGA work, but it seems JCID P15 gained support of reprogramming blank wifi chips for iphones. since they use mostly identical system to M1 platform, this should be available for macs too. 
it is not DIY yourself but it will be doable for advanced mac repairshops who already provide mac ssd upgrades.
